pkg/controller/clone-controller.go

```diff
@@ -768,14 +768,19 @@ func ValidateCanCloneSourceAndTargetContentType(sourcePvc, targetPvc *corev1.Per
 }
 
 // ValidateCanCloneSourceAndTargetSpec validates the specs passed in are compatible for cloning.
-func ValidateCanCloneSourceAndTargetSpec(sourceSpec, targetSpec *corev1.PersistentVolumeClaimSpec, contentType cdiv1.DataVolumeContentType) error {
-	err := ValidateCloneSize(sourceSpec.Resources, targetSpec.Resources)
-	if err != nil {
+func ValidateCanCloneSourceAndTargetSpec(sourcePvc, targetPvc *corev1.PersistentVolumeClaim, contentType cdiv1.DataVolumeContentType) error {
+	err := ValidateCloneSize(sourcePvc.Spec.Resources, targetPvc.Spec.Resources)
+	_, available := targetPvc.Annotations[AnnPermissiveClone]
+	// Due to filesystem overhead differences, the target's size can sometimes be smaller
+	// than the source's one when obtaining said value with the size-detection pod.
+	// In those cases, we use the 'AnnPermissiveClone' annotation to skip the size validation.
```

Comment:
What is the actual scenario for this case? Can we handle it in a better way in the future?

Reply:
When the fs overhead is smaller in the target PVC than in the source PVC (and even when they are the same), the obtained size ends up being smaller than the one specified in the source specs, so I've decided to use this annotation to allow the clone.

I guess this annotation could also be used to fix the already existing bug that happens when cloning from block to filesystem mode.

Reply:
which bug?

Reply:
This one [Bug 2079781](https://bugzilla.redhat.com/show_bug.cgi?id=2079781) (let me know if is inappropriate to share this information here)

Reply:
Which value is reported by the size detection pod? Is it the:

If it is the real virtual image size then the target should have enough space to fit the image plus size overhead. So we can not skip the validation! 

This is probably the only one thing I would like to discsuss in the PR. 
I think @ShellyKa13 found some interesting problems with clone size, so maybe she can help with some ideas?

Reply:
The size detection pod gets the virtual size of the image once it's been resized to match the source PVC - overhead. The problem is that, even after overhead calculations are applied, this image size is usually slightly smaller than the one defined in the source PVC's specs due to the round down performed in this function: https://github.com/kubevirt/containerized-data-importer/blob/6366ef397f2fb5824008ae94a9417bca018c4830/pkg/util/util.go#L453-L459

This is hard to explain here so I'll try using an example: If I create a source PVC with "500Mi" as size and with 5'5% overhead, the size-detection pod doesn't obtain **495452160** bytes as one should've expected, but **494927872** due to the round-down performed in the function below. So, even after reversing this calculation back as we do in the datavolume-controller, the obtained size ends up being smaller than the one in the source PVC's spec. This is why we need a mechanism to enable the clone anyway.

I first tried rounding up the value again for the size comparison, but this behavior happens also if the target's overhead ends up being smaller than the one used in the source, so that's why I used that annotation instead.

Reply:
Ok, that is great explanation, now I get it. 

But it means we might need to improve this or maybe the actual clone with fsOverhead logic, so it is simpler and more predictable. I would like a new issue here on github to "investigate" this case and maybe find ways to simplify/refactor  with your example. So we can act on it in near future.